analyzers/tests/SonarAnalyzer.UnitTest/SymbolicExecution/Roslyn/RoslynSymbolicExecutionTest.cs

```diff
@@ -172,5 +189,89 @@ public void Execute_PersistSymbols_InsideBlock()
+                if (x.Operation.Instance.Kind == OperationKind.Literal)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateCS(@"var a = true; bool b; b = a; Tag(""b"", b);", setter).Collector;
+            collector.Validate("Literal: true", x => x.State[x.Operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.Validate("SimpleAssignment: a = true (Implicit)", x => x.State[new IOperationWrapperSonar(((ISimpleAssignmentOperation)x.Operation.Instance).Target)].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.ValidateTag("b", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void Execute_SimpleAssignmentWithLiteralOnParameters_PropagatesConstraints()
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.Literal)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateCS(@"boolParameter = true; Tag(""boolParameter"", boolParameter);", setter).Collector;
+            collector.Validate("Literal: true", x => x.State[x.Operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.Validate("SimpleAssignment: boolParameter = true", x => x.State[new IOperationWrapperSonar(((ISimpleAssignmentOperation)x.Operation.Instance).Target)].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.ValidateTag("boolParameter", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
+        public void Execute_SimpleAssignmentWithParameterReferenceOnParameters_PropagatesConstraints()
+        {
+            var setter = new PreProcessTestCheck(x =>
+            {
+                if (x.Operation.Instance.Kind == OperationKind.Literal)
+                {
+                    x.State[x.Operation].SetConstraint(DummyConstraint.Dummy);
+                }
+                return x.State;
+            });
+            var collector = SETestContext.CreateCS(@"boolParameter = true; var b = boolParameter; Tag(""b"", b);", setter).Collector;
+            collector.Validate("Literal: true", x => x.State[x.Operation].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.Validate("SimpleAssignment: boolParameter = true", x => x.State[new IOperationWrapperSonar(((ISimpleAssignmentOperation)x.Operation.Instance).Target)].HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+            collector.ValidateTag("b", x => x.HasConstraint(DummyConstraint.Dummy).Should().BeTrue());
+        }
+
+        [TestMethod]
```

Comment:
You can make this `DataTestMethod` to demonstrate all the stuff that can be assignment target and we don't support it. Like `arr[0] = 42`, `dict["key"] = 42`, `other.Property = 42`, `this.Property = 42`, `Sample.StaticProperty = 42`